src/yacs-web/src/components/SelectedCourses.vue

```diff
@@ -0,0 +1,153 @@
+
+export default {
+    name: 'SelectedCourses',
+    mixins: [
+        TimeDateMixin
+    ],
+    props: {
+        courses: Object,
+    },
+    data () {
+        return {
+            ColorService,
+            exportIcon: faPaperPlane,
+            ICS_DAY_SHORTNAMES: ["MO", "TU", "WE", "TH", "FR", "SA", "SU"],
+        }
+    },
+    methods: {
+        removeCourse (course) {
+            this.$emit('removeCourse', course);
+        },
+        toggleCourseSection (course, section) {
+            if (section.selected) {
+                this.$emit('removeCourseSection', section);
+            } else {
+                // Only allow selection of one section per course
+                this.$emit('removeAllCourseSections', course);
+                this.$emit('addCourseSection', course, section);
+            }
+        },
+        exportScheduleToIcs () {
+            let calendarBuilder = window.ics()
+            let semester;
+
+            for (const course of Object.values(this.courses)) {
+                for (const section of course.sections.filter(s => s.selected)) {
+                    for (const session of section.sessions) {
+                        semester = session.semester;
+                        calendarBuilder.addEvent(
+                            `Class: ${course.title}`, 
+                            "LEC day",
+                            session.location,
+                            new Date(`${course.date_start.toDateString()} ${session.time_start}`), 
+                            new Date(`${course.date_start.toDateString()} ${session.time_end}`),
+                            {
+                                freq: "WEEKLY",
+                                interval: 1,
+                                until: course.date_end,
+                                byday: [this.ICS_DAY_SHORTNAMES[session.day_of_week]]
+                            }
+                        );
+                    }
+                }
+            }
+
+            // for (const dayArray of this.schedule.dailySessions) {
+            //     for (const session of dayArray) {
+            //         console.log(session);
+            //         // Add course type in description when available from DB. Add location of session when available.
+            //         let courseInfo = this.courses[session._courseKey];
+            //         semester = session.semester;
+            //         calendarBuilder.addEvent(`Class: ${courseInfo.title}`, "LEC day", session.location, new Date(`${courseInfo.date_start.toDateString()} ${session.time_start}`), new Date(`${courseInfo.date_start.toDateString()} ${session.time_end}`), {
+            //             freq: "WEEKLY",
+            //             interval: 1,
+            //             until: courseInfo.date_end,
+            //             byday: [this.ICS_DAY_SHORTNAMES[session.day_of_week]]
+            //         });
+            //     }
+            // }
```

Comment:
Can remove old implementation.

Originally I had thought each individual schedule would produce an ICS file pertaining to the events of only that subsemester. But grouping all the subsemesters into one ICS file makes sense.